### PR TITLE
Drop Python 2 support (changes in `rtree/index.py` module)

### PR DESCRIPTION
**Pull request by sr-murthy**
* Drop Python 2 support (changes in `rtree/index.py` module)

### CODE REVIEW

**Comment by hobu** [comment]
It looks like tests need to be updated. If there are any more Python 2-specific items, please remove them in this branch. It's time to let it go! ❄️

**Comment by sr-murthy** [comment]
Yes, I've just updated three index test cases that needed to be fixed.

File: rtree/index.py [hunk 1]
```diff
@@ -5,24 +5,12 @@
...
+import pickle
 
 import sys
-if sys.version_info[0] == 2:
-    range = xrange
-    string_types = basestring
-elif sys.version_info[0] == 3:
-    string_types = str
 
+string_types = str
```
Comment (mwtoews) [inline rtree/index.py]: Rather than define this here, on line 223 just replace `string_types` with `str`

Comment (sr-murthy) [reply rtree/index.py]: Makes sense. I've just updated the PR.

**Comment by sr-murthy** [comment]
Closing as changes in this PR have now been incorporated into the Flake8 linting PR, which I've just merged.

**Pull request closed without merging**